src/protocol.rs

```diff
@@ -313,42 +324,60 @@ impl std::fmt::Display for ParticipantIdentifier {
+/// Session identifiers have two roles in the protocol: they tag messages
+/// and they are incorporated as context into zero-knowledge proofs.
+/// The "globally unique" property allows participants to distinguish
+/// messages belonging to different, concurrent protocol runs,
+/// prevents collisions between messages belonging to
+/// different sessions, and prevents replay attacks by associating messages and
+/// zero-knowledge proofs
+/// with the session, fixed parameters, and previous subprotocols to which
+/// they correspond.
 ///
 /// # Discrepancies with the paper with respect to session identifiers:
+/// The paper defines session identifiers, denoted `sid` and `ssid`, somewhat
+/// differently that we implement them in this codebase. We believe this
+/// implementation is functionally equivalent to what the paper describes.
+///
+/// 1. The paper incorporates many types of data into its session and
+/// sub-session identifiers, including fixed parameters, the participant set,
+/// and outputs from previous protocols; these identifiers are used both to tag
+/// messages and incorporate context into proofs. The codebase defines a single
+/// `Identifier` type that is expected to be chosen uniformly at random; this is
+/// used to tag messages. The parameters, participants, and outputs (as well as
+/// the `Identifier`) are incorporated into proofs using a different mechanism
+/// to define the proof context.
 ///
-/// Discrepancy (A): The paper distinguishes between the Session identifiers
-/// (sid) which are created from shared parameters in the protocol at the
-/// beginning of key generation and the Sub-Session identifiers (ssid) that are
-/// created from sid and other post key generation shared parameters for the
-/// purpose of pre/signing. The codebase does not make this kind of distinction
-/// relying instead on a single type for all session identifiers. The
-/// distinction between sessions and sub-sessions is inherently enforced by the
-/// order of inputs and outputs to different stages in the protocol.
+/// 2. The paper distinguishes between sessions (keygen) and sub-sessions
+/// (auxinfo and presign).
+/// The codebase requires the calling application to select a new, unique
+/// session [`Identifier`]s at three points:
+/// (1) immediately before starting a new keygen session;
+/// (2) immediately before starting a new auxinfo session;
+/// (3) immediately before starting a new presigning session.
 ///
-/// Discrepancy (B): The codebase  instantiates [`Identifier`]s in three
-/// different ways: (1) as a session identifier for keygen, that are created at
-/// the start of a key generation instance; (2) as a session identifier for
-/// pre-signing, (3) as a session identifier for auxiliary information
-/// generation sessions; The paper itself only distinguishes between sessions
-/// and sub-sessions and combine (2) and (3).
+///     In fact, we think this is a minor bug in Figure 6 of the paper, since
```

Comment:
switch "this" to "there"

Reply:
It's a bug. `ssid` contains the values you're generating by running AuxInfo.

See also Figure 3, where `sid` is passed as the session identifier input to Auxinfo (which makes way more sense).

Reply:
Fixed. Also moved this to a footnote because it's more of a point of interest than something the reader needs to take action on.